fendermint/actors/blobs/shared/src/state.rs

```diff
@@ -415,6 +430,100 @@ impl fmt::Display for TtlStatus {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize_tuple, Deserialize_tuple)]
+pub struct CreditApprovals {
+    pub root: hamt::Root<Address, CreditApproval>,
+    size: u64,
+}
+
+impl CreditApprovals {
+    pub fn new<BS: Blockstore>(store: &BS) -> Result<Self, ActorError> {
+        let root = hamt::Root::<Address, CreditApproval>::new(store, "credit_approvals")?;
+        Ok(Self { root, size: 0 })
+    }
+
+    pub fn hamt<BS: Blockstore>(
+        &self,
+        store: BS,
+    ) -> Result<hamt::map::Hamt<BS, Address, CreditApproval>, ActorError> {
+        self.root.hamt(store, self.size)
+    }
+    pub fn save_tracked(
+        &mut self,
+        tracked_flush_result: TrackedFlushResult<Address, CreditApproval>,
+    ) {
+        self.root = tracked_flush_result.root;
+        self.size = tracked_flush_result.size
+    }
+
+    pub fn len(&self) -> u64 {
+        self.size
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.size == 0
+    }
+}
+
+/// The return type used for Account.
+#[derive(Debug, Serialize_tuple, Deserialize_tuple)]
+pub struct AccountInfo {
```

Comment:
I created this new type so that the actor can return the account info using `HashMap` for `approvals_to` and `approvals_from`  as before. I wasn't sure how clients would handle if we returned the hamt instead

This is used in [`buy_credit`](https://github.com/recallnet/ipc/pull/541/files#diff-a3f97a7276b9e6bd346858398b5297de5aeb01f30473e2d94aec44b13b6f9af0R84), and [`get_account`](https://github.com/recallnet/ipc/pull/541/files#diff-a3f97a7276b9e6bd346858398b5297de5aeb01f30473e2d94aec44b13b6f9af0R305), and [here](https://github.com/recallnet/ipc/pull/541/files#diff-732ecaf2caff3ea98ad38ab0d514481f98a542238bdd0a10faded80f1b6f1941R492-R524) we convert an `account` to `account_info` resolving the addresses

The SDK must be updated to account for this new return type